src/odemis/driver/test/hamamatsurx_test.py

```diff
@@ -108,6 +115,96 @@ def tearDownClass(cls):
+                1.e-9: 7.99e-9,
+                2.e-9: 9.63e-9,
+                5.e-9: 33.2e-9,
+                10.e-9: 45.9e-9,
+                20.e-9: 66.4e-9,
+                50.e-9: 102e-9,
+                100.e-9: 169e-9,
+                200.e-9: 302e-9,
+                500.e-9: 731e-9,
+                1.e-6: 1.39e-6,
+                2.e-6: 2.69e-6,
+                5.e-6: 7.02e-6,
+                10.e-6: 13.8e-6,
+                20.e-6: 26.7e-6,
+                50.e-6: 81.6e-6,
+                100.e-6: 161e-6,
+                200.e-6: 320e-6,
+                500.e-6: 798e-6,
+                1.e-3: 1.62e-3,
+                2.e-3: 3.18e-3,
+                5.e-3: 7.88e-3,
+                10.e-3: 15.4e-3,
+            }
+        })
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.streakcam.terminate()
+
+    def test_error(self):  # TODO more testcases
+        """Test the different RemoteEx errors possible."""
+        # send wrong command, will timeout
+        with self.assertRaises(util.TimeoutError):
+            self.streakcam.sendCommand("Appinfoo", "type")
+
+    ### Delay generator #####################################################
+    def test_TriggerDelay(self):
+        """Test Acquisition Mode VA for delay generator."""
+        self.delaybox.triggerDelay.value = 0
+        prev_triggerDelay = self.delaybox.triggerDelay.value
+        # change trigger delay VA
+        self.delaybox.triggerDelay.value = 0.000001  # 1us
+        cur_triggerDelay = self.delaybox.triggerDelay.value
+        # check previous and current value are not the same
+        self.assertNotEqual(prev_triggerDelay, cur_triggerDelay)
+        # get trigger delay from hardware
+        remoteEx_triggerDelay = self.delaybox.GetTriggerDelay()
+        self.assertEqual(cur_triggerDelay, remoteEx_triggerDelay)
+
+        # request value, which is not in range of VA
+        with self.assertRaises(IndexError):
+            self.delaybox.triggerDelay.value = -1
+
+    def test_TimeRange(self):
+        """Test time range VA for sweeping of streak unit."""
+        for timeRange in self.streakunit.timeRange.choices:  # values different from yaml due to floating point issues
+            # change timeRange VA to value in range
+            self.streakunit.timeRange.value = timeRange
+            self.assertAlmostEqual(self.streakunit.timeRange.value, timeRange)
+
+            tr2d = self.delaybox._metadata.get(model.MD_TIME_RANGE_TO_DELAY)
+            if tr2d:
+                key = util.find_closest(timeRange, tr2d.keys())
+                # check that the corresponding trigger delay is set when changing the .timeRange VA
+                md_triggerDelay = tr2d[key]
+                self.assertAlmostEqual(self.delaybox.triggerDelay.value, md_triggerDelay)
+
+        # request value, which is not in choices of VA
+        with self.assertRaises(IndexError):
+            self.streakunit.timeRange.value = 0.000004  # 4us
+
```

Comment:
Add a test case for delays B to H.
(Maybe, skip it if the VAs are missing, so that it works with the synchroscan delay unit)